### PR TITLE
Refactor/query builder v2

### PR DESCRIPTION
**Pull request by KristianVepsalainen**
## Context

Progress update on the query builder refactor discussed earlier — 
not ready to merge, opening as draft for feedback before I invest 
more time in polish.

## What's done

- All 25 `include_*` fields and all 10 `resource_type` filters migrated 
  to a data-driven structure (`field_specs` list + a single loop that 
  builds SELECT/WHERE/GROUP BY together)
- `aggregate_vars` parameter works generically — pass any aggregatable 
  field name(s) and it handles `GROUP_CONCAT` + `GROUP BY` automatically, 
  including multi-variable fields like `lbs` (tested: all three of 
  `?lbs`, `?lbcelex`, `?lbsuffix` aggregate correctly together)
- Built a byte-identity test harness comparing `elx_make_query_new()` 
  against `elx_make_query()` — 44 test combinations covering every 
  field individually, several combinations, all resource types, 
  `date_from`/`date_to`, `order = TRUE`, and edge cases like the 
  `directory`/`directory_code` shared-variable situation. All 44 pass.

## What's not done yet

- Roxygen2 documentation for `elx_make_query_new()`
- A proper testthat test suite (the byte-identity harness is a dev 
  tool, not a package test)
- Haven't touched naming/export decisions — currently unexported, 
  function name is a placeholder

## Question

Does this direction look right to you? If so, I'll go ahead and get 
the documentation and testthat suite in shape next. Happy to adjust 
the approach first if something looks off.

### CODE REVIEW

File: R/elx_make_query_new.R [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+  if (use_group_by) {
+    query <- paste(query, "GROUP BY", paste(group_vars, collapse = " "))
+  }
+  
+  if (order == TRUE){
+    order_var <- if (use_group_by) "?work" else "?date"
+    query <- paste0(query, " order by str(", order_var, ")")
+  }
+  
+  if (!is.null(limit) & is.integer(as.integer(limit))){
```
Comment (michalovadek) [inline R/elx_make_query_new.R]:
```suggestion
# is.integer can wrongly validate an integer NA. Maybe:

limit_int <- suppressWarnings(as.integer(limit))
if (!is.null(limit) && !is.na(limit_int)) {
  query <- paste(query, "limit", limit_int)
}
```

**Comment by KristianVepsalainen** [comment]
Hi Michal,

Just wanted to follow up on this draft PR whenever you have a chance.

I'm happy to continue with the documentation and the `testthat` suite, but before investing more time I wanted to check whether the overall direction of this refactor looks good to you, or whether you'd prefer a different approach.

No rush if you've been busy — I'd just like to know whether I should keep moving this forward.

Thanks!

**Comment by michalovadek** [comment]
yes, I think it is on the right track - if you could address the comments I left in the review we could move it forward

**Comment by KristianVepsalainen** [comment]
Thanks! I don't see the review comments on my end — could you double check they went through / were submitted? GitHub sometimes leaves pending review comments unpublished. Happy to address them once I can see them. 

And sorry about pressing that "ready for review" button. It was mistake.

**Review by michalovadek — CHANGES_REQUESTED** [review]

**Comment by michalovadek** [comment]
> Thanks! I don't see the review comments on my end — could you double check they went through / were submitted? GitHub sometimes leaves pending review comments unpublished. Happy to address them once I can see them.
> 
> And sorry about pressing that "ready for review" button. It was mistake.

my bad! I forgot to press submit for review

**Comment by KristianVepsalainen** [comment]
Addressed all three review comments:

Added explanatory comments on the directory/directory_code shared-variable design
Fixed the limit validation to guard against NA from non-numeric input, using your suggested approach
Added date_to >= date_from validation

Also added roxygen2 documentation (kept @noRd for now since export/naming wasn't decided) and a proper testthat suite — 23 tests covering byte-identity against the old builder, aggregate_vars behavior (including the multi-variable lbs case), and the new validation logic.

Full devtools::check(cran = TRUE): 0 errors, 0 warnings.

Ready for another look whenever you have time — let me know if anything else needs adjusting.

**Comment by KristianVepsalainen** [comment]
Fixed — the CI failure was a side effect of the merge picking up your recent include_author change (auto-aggregating without needing aggregate_vars), which my byte-identity test hadn't accounted for. Updated the test to reflect this as an intentional, documented behavior difference between the two functions (elx_make_query auto-aggregates author; elx_make_query_new requires explicit aggregate_vars = "author").

All 6 CI checks pass now. Ready for another look whenever convenient.

**Comment by michalovadek** [comment]
can we get the test comments in English please?

**Comment by KristianVepsalainen** [comment]
Apologies for that — those comments were leftover from some earlier work I'd done on a different, Finnish-language project, and I must have carried the habit over without noticing. Finnish is actually my native language, English isn't, so these things sometimes slip through. Translated all of them now — should be all English throughout the test file.

**Comment by michalovadek** [comment]
thanks. I think there are a few ones left in the other changed files as well

**Comment by KristianVepsalainen** [comment]
Found and translated the remaining ones — should now be all English across elx_make_query_new.R, elx_query_helpers.R, and test_byte_identical.R. Let me know if anything else needs a pass.

**Comment by michalovadek** [comment]
just want to check something -- should this line:

https://github.com/KristianVepsalainen/eurlex/blob/463450dae07c19064bc0ea67a894f5faa68fd2d6/R/elx_make_query_new.R#L407

be outside the preceding `else` statement?

**Comment by KristianVepsalainen** [comment]
Good catch — that's actually mirroring the old elx_make_query()'s existing behavior (when both include_directory and include_directory_code are TRUE, it also adds both WHERE clauses redundantly, just without the SELECT duplication). It's harmless functionally since both OPTIONAL blocks target the same ?directory variable, but it is redundant.

Since we're already deviating from strict byte-identity in a few places per your other review comments, want me to fix this properly — skip directory_code's WHERE clause too when directory is already active, not just its SELECT variable?
